src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -349,56 +349,82 @@
                        :value index-name})))
             set)))))
 
-(def ^:dynamic *nested-field-column-max-row-length*
-  "Max string length for a row for nested field column before we just give up on parsing it.
-  Marked as mutable because we mutate it for tests."
-  50000)
-
-(defn- flattened-row [field-name row]
-  (letfn [(flatten-row [row path]
-            (lazy-seq
-              (when-let [[[k v] & xs] (seq row)]
-                (cond (and (map? v) (not-empty v))
-                      (into (flatten-row v (conj path k))
-                            (flatten-row xs path))
-                      :else
-                      (cons [(conj path k) v]
-                            (flatten-row xs path))))))]
-    (into {} (flatten-row row [field-name]))))
+(def ^:const max-nested-field-columns
+  "Maximum number of nested field columns."
+  100)
 
 (def ^:private ^{:arglists '([s])} can-parse-datetime?
   "Returns whether a string can be parsed to an ISO 8601 datetime or not."
   (mr/validator ::lib.schema.literal/string.datetime))
 
 (defn- type-by-parsing-string
   "Mostly just (type member) but with a bit to suss out strings which are ISO8601 and say that they are datetimes"
-  [member]
-  (let [member-type (type member)]
-    (if (and (instance? String member)
-             (can-parse-datetime? member))
-      java.time.LocalDateTime
-      member-type)))
-
-(defn- row->types [row]
-  (into {} (for [[field-name field-val] row
-                 ;; We put top-level array row type semantics on JSON roadmap but skip for now
-                 :when (map? field-val)]
-             (let [flat-row (flattened-row field-name field-val)]
-               (into {} (map (fn [[k v]] [k (type-by-parsing-string v)]) flat-row))))))
-
-(defn- describe-json-xform [member]
-  ((comp (map #(for [[k v] %
-                     :when (< (count v) *nested-field-column-max-row-length*)]
-                 [k (json/parse-string v)]))
-         (map #(into {} %))
-         (map row->types)) member))
-
-(def ^:const max-nested-field-columns
-  "Maximum number of nested field columns."
-  100)
+  [value]
+  (if (and (string? value)
+           (can-parse-datetime? value))
+    java.time.LocalDateTime
+    (type value)))
+
+(defn- json-parser ^JsonParser [v]
+  (let [f (JsonFactory.)]
+    (if (string? v)
+      (.createParser f ^String v)
+      (.createParser f ^java.io.Reader v))))
+
+(defn- number-type [t]
+  (u/case-enum t
+    JsonParser$NumberType/INT Integer
+    JsonParser$NumberType/LONG Long
+    JsonParser$NumberType/FLOAT Float
+    JsonParser$NumberType/DOUBLE Double
+    JsonParser$NumberType/BIG_DECIMAL BigDecimal
+    JsonParser$NumberType/BIG_INTEGER clojure.lang.BigInt))
```

Comment:
Why do we care about exposing `Integer` and `Float`? We could be opinionated like Clojure and just promote them.

Reply:
Hmmm, maybe; I've tried to repeat whatever behavior we had in place and was just eager to finally get PR up. 😁 I'll look into making this simpler.